cockatrice/src/pilezone.cpp

```diff
@@ -44,7 +44,15 @@ void PileZone::paint(QPainter *painter, const QStyleOptionGraphicsItem * /*optio
 void PileZone::addCardImpl(CardItem *card, int x, int /*y*/)
 {
     connect(card, SIGNAL(sigPixmapUpdated()), this, SLOT(callUpdate()));
-    cards.insert(x, card);
+
+    if (0 <= x) {
```

Comment:
to reiterate on my previous comment here, this is not where we should be doing this check, we should be doing it in CardZone::addCard before we call this function which is implemented in all other card zones, that way the issue will not happen in those classes as well, alternatively also add this in all those other zones

Reply:
That makes sense. I've redone it to do the check upstream